src/app/app.module.ts

```diff
@@ -91,68 +94,93 @@ export function appInitializer(
 
           // Set this back so that second flow through we do the proper code flow to get a refresh token.
           sessionStorage.setItem('syndesis-first-idp', 'false');
-          oauthService.hybrid = originalHybrid;
 
-          let autoLinkGithHub = configService.getSettings('oauth')[
-            'auto-link-github'
-          ];
-          if (autoLinkGithHub === undefined) {
-            autoLinkGithHub = true;
-          }
+          /**
+           * Redirects user to setup page if GitHub setup is still pending
+           */
+          const apiEndpoint = configService.getSettings().apiEndpoint;
+          const accessToken = oauthService.getAccessToken();
+          setupService.isSetupPending(apiEndpoint, accessToken)
+            .then(function (setupPending) {
+              if (setupPending) {
+                window.location.assign('/setup');
+              } else {
+                oauthService.hybrid = originalHybrid;
 
-          // If this wasn't the autolink flow then rekick off flow with state set to autolink.
-          if (autoLinkGithHub && oauthService.state !== 'autolink') {
-            // Client suggested IDP works great with Keycloak.
-            oauthService.loginUrl += '?kc_idp_hint=github';
-            // Clear session storage before trying again.
-            oauthService.logOut(true);
-            // And kick off the login flow again.
-            return oauthService.initImplicitFlow('autolink');
-          }
+                let autoLinkGithHub = configService.getSettings('oauth')[
+                  'auto-link-github'
+                ];
+                if (autoLinkGithHub === undefined) {
+                  autoLinkGithHub = true;
+                }
+
+                // If this wasn't the autolink flow then rekick off flow with state set to autolink.
+                if (autoLinkGithHub && oauthService.state !== 'autolink') {
+                  // Client suggested IDP works great with Keycloak.
+                  oauthService.loginUrl += '?kc_idp_hint=github';
+                  // Clear session storage before trying again.
+                  oauthService.logOut(true);
+                  // And kick off the login flow again.
+                  return oauthService.initImplicitFlow('autolink');
+                }
+
+                init2(oauthService, userService, ngZone, notificationService);
```

Comment:
So for this kind of thing where `init2` is only going to be called from this area I'm a bigger fan of just using an inner function, then the relationship between this code and the function being called is a little easier to follow.